### PR TITLE
Add support for new Manifest.toml format during code load

### PR DESCRIPTION
**Pull request by IanButterworth**
Initial discussion in https://github.com/JuliaLang/Pkg.jl/pull/2557

First Pkg PR  https://github.com/JuliaLang/Pkg.jl/pull/2561

- [x] Needs tests for loading from the new manifest format

### CODE REVIEW

**Review by Rada8732 — APPROVED** [review]

File: base/loading.jl [hunk 1]
```diff
@@ -485,12 +485,16 @@ function explicit_project_deps_get(project_file::String, name::String)::Union{No
     return nothing
 end
 
+# handles returning deps list for both old and new manifest formats
+# if no `deps` field, then its either the older flat format or empty, so return self
+get_deps(d) = get(d, "deps", d)
```
Comment (IanButterworth) [inline base/loading.jl:490]: Perhaps also check if `deps` has children that look like packages

File: base/loading.jl [hunk 2]
```diff
@@ -498,12 +498,34 @@ function explicit_project_deps_get(project_file::String, name::String)::Union{No
...
+    end
+end
+
+# returns a deps list for both old and new manifest formats
+function get_deps(raw_manifest::Dict)
+    if is_v1_format_manifest(raw_manifest)
+        return raw_manifest
+    else
+        # if the manifest has no deps, there won't be a `deps` field
+        return get(raw_manifest, "deps", Dict{String, Any}())
```
Comment (KristofferC) [inline base/loading.jl:519]:
```suggestion
        return get(Dict{String, Any}, raw_manifest, "deps")
```

**Review by KristofferC — APPROVED** [review]

File: base/loading.jl (hunk 2 above)
Comment (IanButterworth) [reply base/loading.jl]: Huh, TIL

**Comment by DilumAluthge** [comment]
I've added this to the 1.7 milestone BTW.

**Comment by StefanKarpinski** [comment]
We had also discussed backporting support for the new manifest format to the 1.6 series since it will be the new LTS and we want moving back and forth to not become too annoying. Should this be backported to that branch as well?

**Pull request merged**